cli/src/main/java/com/devonfw/tools/ide/context/GitContext.java

```diff
@@ -2,93 +2,156 @@
    *
    * @param gitRepoUrl the git remote URL to clone from.
    * @param targetRepository the {@link Path} to the target folder where the git repository should be cloned or pulled.
-   * It is not the parent directory where git will by default create a sub-folder by default on clone but the * final
-   * folder that will contain the ".git" subfolder.
+   *        It is not the parent directory where git will by default create a sub-folder by default on clone but the
+   *        final folder that will contain the ".git" subfolder.
+   * @throws CliOfflineException if offline and cloning is needed.
    */
   void pullOrClone(String gitRepoUrl, Path targetRepository);
 
   /**
    * Runs a git pull or a git clone.
    *
    * @param gitRepoUrl the git remote URL to clone from.
-   * @param branch the branch name e.g. master.
+   * @param branch the explicit name of the branch to checkout e.g. "main" or {@code null} to use the default branch.
    * @param targetRepository the {@link Path} to the target folder where the git repository should be cloned or pulled.
-   * It is not the parent directory where git will by default create a sub-folder by default on clone but the * final
-   * folder that will contain the ".git" subfolder.
+   *        It is not the parent directory where git will by default create a sub-folder by default on clone but the
+   *        final folder that will contain the ".git" subfolder.
+   * @throws CliOfflineException if offline and cloning is needed.
    */
   void pullOrClone(String gitRepoUrl, String branch, Path targetRepository);
 
   /**
-   * Runs a git clone. Throws a CliException if in offline mode.
+   * Runs a git clone.
    *
    * @param gitRepoUrl the {@link GitUrl} to use for the repository URL.
    * @param targetRepository the {@link Path} to the target folder where the git repository should be cloned or pulled.
-   * It is not the parent directory where git will by default create a sub-folder by default on clone but the * final
-   * folder that will contain the ".git" subfolder.
+   *        It is not the parent directory where git will by default create a sub-folder by default on clone but the *
+   *        final folder that will contain the ".git" subfolder.
+   * @throws CliOfflineException if offline and cloning is needed.
    */
   void clone(GitUrl gitRepoUrl, Path targetRepository);
 
   /**
    * Runs a git pull.
    *
    * @param targetRepository the {@link Path} to the target folder where the git repository should be cloned or pulled.
-   * It is not the parent directory where git will by default create a sub-folder by default on clone but the * final
-   * folder that will contain the ".git" subfolder.
+   *        It is not the parent directory where git will by default create a sub-folder by default on clone but the *
+   *        final folder that will contain the ".git" subfolder.
    */
   void pull(Path targetRepository);
 
   /**
-   * Runs a git reset if files were modified.
```

Comment:
Actually I removed the "if files where modified" because I did not understand why it was explicitly mentioned.
JavaDoc should rather explain what the outcome of a method is rather than how it is implemented.
The outcome is that local changes are reverted. Whether we first check if we can skip calling `git reset` or not is an implementation detail that may change at any time (and actually I do not know why where are doing it this way since `git reset` should not do anything if there are no local changes).
However, I followed your suggestion to get this PR completed. Maybe it might help some developer this way - who knows...